### DISCUSSION TITLE
Does setValue trigger a re-render?

### DISCUSSION THREAD

**Discussion by rossthedevigner** [react-hook-form/react-hook-form#10362] [Q&A]
Simple question (I think), but does `setValue` trigger a re-render? If not, is there anything specific to using it with `...register()` that it would?

**Comment by Moshyfawn** [comment]
`setValue` doesn't trigger a re-render of your component on it's own. In React, you need a reactive property _(state)_ to change to signal React to re-render your component.

`register` is an uncontrolled input API, which means, the value of the input is, well, the input itself; that means that there's no reactive property for React to reference in order to determine when to re-render your component.

That said, calling `setValue` should update the specified input value:
1. Open [this sandbox](https://codesandbox.io/s/sleepy-einstein-wmk244?file=%2Fsrc%2FApp.tsx)
2. Type anything into the "Set name" input
3. See the "Name" input value being updated _(no component re-render)_
